### DISCUSSION TITLE
Connect with duplicates

### DISCUSSION THREAD

**Discussion by patrykgulas** [prisma/prisma#16156] [Q&A] [answered]
Hey 👋 

In my app I have the following schema:

```prisma
// schema.prisma

model Post {
  id String @id
  name String
  tags Tag[]
}

model Tag {
  id String @id
  label String
  posts Post[]
}
```

**I would like to `connect` many tags to `Post` with possible duplicates.** 
For example, I would like to get in return `Post` with the following structure:

```json
{
  "id": "POST_1_ID",
  "name": "Post title",
  "tags": ["Tag1", "Tag1", "Tag2"]
}
```

To seed data I use the following script logic:

```ts
// seed.ts

await prisma.tag.createMany({
  data: [{
    id: 'TAG_1_ID',
    label: 'Tag 1'
  }, {
    id: 'TAG_2_ID',
    label: 'Tag 2'
  }]
})

await prisma.post.create({
  data: {
    id: 'POST_1_ID',
    name: 'Post title',
    tags: {
      connectOrCreate: [{
        where: { id: 'TAG_1_ID' },
        create: {
          id: 'TAG_1_ID',
          name: 'Tag 1'
        }
      }, {
        where: { id: 'TAG_1_ID' },
        create: {
          id: 'TAG_1_ID',
          name: 'Tag 1'
        }
      }, {
        where: { id: 'TAG_2_ID' },
        create: {
          id: 'TAG_2_ID',
          name: 'Tag 2'
        }
      }]
    }
  }
})
```

The result is that `_PostTags` table in the database is getting only two rows saved:

| A | B |
|---|---|
| POST_ID_1 | TAG_ID_1 | 
| POST_ID_1 | TAG_ID_2 |

Do you maybe have an idea what should be a solution to such case and correct design? 🤔

**Comment by pantharshit00** [comment, marked as answer]
Hey @patrykgulas 

You can supplying same ID in second `createOrConnect` item so it won't actually create a duplicate. You will need to use a different ID if you want the call to create a new Tag with same label.

**Reply by patrykgulas** [reply]
@pantharshit00 thanks for the reply 👌 

For now, I resolved my case by adding PostTag table which includes also quantity.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
